DigitalLearningSolutions.Data.Tests/DataServices/GroupsDataServiceTests.cs

```diff
@@ -157,6 +157,31 @@ public async Task RemoveRelatedProgressRecordsForGroupDelegate_updates_progress_
             }
         }
 
+        [Test]
+        public async Task RemoveRelatedProgressRecordsForGroupDelegate_does_not_update_progress_record_when_course_is_shared_by_another_group()

```

Comment:
NOTE TO SELF: this test requires a new record adding to the test DBs. When this is about to get merged, add the record to the test environment DBs and let everyone know in the internal chat.

Reply:
It is indeed just this test that requires the new data. There is a circumstance that could arise when a progress record is shared by multiple courses in different groups that one person is on, but we don't have any data that actually fits that category. 

It would be feasible to add this data here, but it would require using three of the new data service methods created as part of this ticket just to set up the record. Correct me if I'm wrong, but I feel like in the past we've only ever used simple tests for the data services, and whenever new data is needed we add them in scripts to prevent having a test that relies on other methods (that also need tests). But it should be simple enough to add it here if we do want that.

Reply:
If you think using the data services to set up the data would be detrimental to the quality of the test (I think I'd agree) I think it's better to use the sql script you've created to set up the data - but at the start of the test rather than as a permanent change to the db.

There are a few downsides to adding permanent test data:
* It requires everyone on the team to run another sql script / adds another script to run when setting up the test db
* It adds another entity to possibly clash with when writing other data service tests.
* It means you need more contextual information to understand why the test works.
* It makes the test setup much harder to change, should that be required in the future.

So I think it's good to minimise it as much as possible.